src/commands/math/commands.js

```diff
@@ -1007,70 +1007,62 @@ Environments.matrix = P(Environment, function(_, super_) {
     table.toggleClass('mq-rows-1', table.find('tr').length === 1);
     this.relink();
   };
-  _.upInto = function(dir, cursor) {
-    var rowSize = this.getRowSize();
-    if (cursor.parent.parent === this) {
-      // We're inside the matrix.
-      var i = this.blocks.indexOf(cursor.parent);
+  // Enter the matrix at the top or bottom row if updown is configured.
+  _.getEntryPoint = function(dir, cursor, updown) {
+    if (updown === 'up') {
       if (dir === L) {
-        // If we're on the left edge and moving left, we should exit.
-        return i % rowSize === 0;
+        return this.blocks[this.rowSize - 1];
       } else {
-        // If we're on the right edge and moving right, we should exit.
-        return (i + 1) % rowSize == 0;
+        return this.blocks[0];
+      }
+    } else if (updown === 'down') {
+      // Otherwise, we must be about to enter the matrix.
+      if (dir === L) {
+        return this.blocks[this.blocks.length - 1];
+      } else {
+        return this.blocks[this.blocks.length - this.rowSize];
       }
-    }
-    // Otherwise, we must be about to enter the matrix.
-    if (dir === L) {
-      return this.blocks[rowSize - 1];
     } else {
-      return this.blocks[0];
+      pray("Invalid value for updown '" + updown + "'", false);
```

Comment:
Hey @alexf101, I think this check is unnecessary because the value is already [validated in optionProcessors](https://github.com/Learnosity/mathquill/blob/fb4e7e2e84432f539ccd5d10ff37d94f9fb61e8e/src/services/keystroke.js#L168-L174
).

And it looks like `getEntryPoint` can return undefined without causing any problems.

Reply:
Ah, so it is! I'll remove the branch.